src/directory/managed_directory.rs

```diff
@@ -222,9 +223,21 @@ impl ManagedDirectory {
             .write()
             .expect("Managed file lock poisoned");
         let has_changed = meta_wlock.managed_paths.insert(filepath.to_owned());
-        if has_changed {
-            save_managed_paths(self.directory.as_ref(), &meta_wlock)?;
+        if !has_changed {
+            return Ok(());
+        }
+        save_managed_paths(self.directory.as_ref(), &meta_wlock)?;
+        if meta_wlock.managed_paths.len() > 1 {
+            // This is not the first file we add.
+            // `.managed.json` has been already properly created and we do not need to
+            // sync its parent directory.
```

Comment:
I don't understand why, the existence of .managed.json is important, but not its content?

Reply:
I am not sure what you read here. We do fsync its content today upon each write.
However, we do not flush the parent directory every time we update it.